specification/under_development/profiles/Base_Profile.md

```diff
@@ -0,0 +1,484 @@
+
+To execute a given bitcode, the backend needs to know how to process qubit and
+result pointers used by a program. This is achieved by storing metadata in the
+form of [module flags](#module-flags-metadata) in the bitcode. QIR does not make
+a type distinction or for example uses a different address space for the two
+kinds of pointers. This ensures that libraries and optimization passes that map
+between different instruction sets do not need to distinguish whether the
+compiled application code makes use of dynamic qubit and result management or
+not.
+
+To be compliant with the Base Profile specification, the program must not make
+use of dynamic qubit or result management; instead, qubits and results must be
+identified by a constant integer value that is bitcast to a pointer to match the
+expected type. How such an integer value is interpreted and specifically how it
+relates to hardware resources is ultimately up to the executing backend.
+
+Additionally, the Base Profile imposes the following restrictions on qubit and
+result usage:
+
+- Qubits must not be used after they have been measured. Which quantum
+  instructions perform measurements is defined in the specification for known
+  [quantum instructions](../Instruction_Set.md). For quantum instructions that
+  are not listed there, we refer to the documentation provided by the targeted
+  backend that supports them.
+
+- Results can only be used either as `writeonly` arguments, or as arguments to
+  [output recording functions](#output-recording). We refer to the [LLVM
+  documentation](https://llvm.org/docs/LangRef.html#function-attributes)
+  regarding how to use the `writeonly` attribute.
+
+## Quantum Instruction Set
+
+For a quantum instruction set to be fully compatible with the Base Profile, it
+must satisfy the following two requirements:
+
+- All functions must return void; the Base Profile does not permit to call
+  functions that return a value. Functions that measure qubits must take the
+  qubit pointer(s) as well as the result point(s) as arguments.
+
+- Parameters of type `%Result*` must be `writeonly` parameters; only the runtime
+  function `__quantum__rt__result_record_output` used for [output
+  recording](#output-recording) may read a measurement result.
+
+For more information about the relation between a profile specification and the
+quantum instruction set we refer to the paragraph on [Bullet 1](#base-profile)
+in the introduction of this document. For more information about how and when
+the QIS is resolved, as well as recommendations for front- and backend
+developers, we refer to the document on [compilation stages and
+targeting](../Compilation_And_Targeting.md).
+
+## Output Recording
+
+The program output of a quantum application is defined by a sequence of calls to
+runtime functions that return values produced by the computation. In the case of
+the Base Profile, these calls are contained within the last block of the entry
```

Comment:
"the last block" => "the second (and last) block" ?

Reply:
I think that is made clear enough above, and  if there are three blocks then it is in the third block - so saying the last block seems like the relevant information here. If you don't mind, I'll leave it as is.